src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
1.  Only use `let` if it's a must (in most cases using `const` should be enough)
2.  In this task we are learning to use array methods, so why don't we replace `forEach` with `reduce`? (read about it [here](https://www.w3schools.com/jsref/jsref_reduce.asp))
3. How it works in code bellow:
- use map on men array men to return an array of numbers (men's ages)
- use reduce to sum up all ages

```suggestion
  const menTotalAge = men.map(man => man.age).reduce(function(sum, number) {
        return sum + number;
    });

const AverageAge = menTotalAge / men.length

  return AverageAge;
```